examples/rust-sysvar/src/processor.rs

```diff
@@ -0,0 +1,37 @@
+//! Program instruction processor
+
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    info,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvar::{self, clock::Clock, Sysvar},
+};
+
+/// Instruction processor
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    _instruction_data: &[u8],
+) -> ProgramResult {
+    // Create in iterator to safety reference accounts in the slice
+    let account_info_iter = &mut accounts.iter();
+
+    // As part of the program specification the first account is the clock
+    // sysvar
+    let clock_sysvar_info = next_account_info(account_info_iter)?;
+
+    if *clock_sysvar_info.key != sysvar::clock::id() {
+        // first account is not the clock sysvar
+        return Err(ProgramError::InvalidArgument);
```

Comment:
`Clock::from_account_info()` now performs this check right?   So this condition can be removed?

Reply:
true dat